nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/field/ObjectLocalDateTimeFieldConverter.java

```diff
@@ -85,13 +90,31 @@ public LocalDateTime convertField(final Object field, final Optional<String> pat
                     return tryParseAsNumber(string, name);
                 }
             }
+            case Object[] objectArray -> {
+                if (objectArray.length != INT96_TIMESTAMP_BYTES_LENGTH) {
```

Comment:
This is a generalized converter, so introducing Parquet-specific handling does not look like the right approach. The implementation should be localized to the ParquetReader.